### ISSUE TITLE
[Feature request] Indicate incomplete visibility of element type

### ISSUE THREAD

**Issue by sdasda7777** [OpenPonk/openponk#91]
I've noticed that the TypeShowHidePresenter is pretty misleading: it shows checkmark for every type where at least one item is visible, even if some are hidden.

On the following picture there are two associations, one of them being hidden, yet the TypeShowHidePresenter makes it seem like all should be visible.

![image](https://user-images.githubusercontent.com/17746796/214497952-e9625acf-8197-4ce9-be7d-0464c9d27065.png)

The behaviour I would more hope for would be using a tristate checkbox, where checked state would only be shown if all elements are visible, unchecked only if none are visible, and the third state otherwise:

![image](https://user-images.githubusercontent.com/17746796/214498320-cf609c5e-60ad-4791-924b-29fe37085d2f.png)

**Comment by JanBliznicenko** [comment]
Yes, it is misleading. Unfortunately, this is limitation of current implementation of Spec, Morphic and Athens. It would be awesome to have it, but it would take quite a lot of effort I think.
It would probably need to change or create special case of SpCheckBoxTableColumn, SpMorphicTableColumnVisitor, SpMorphicTableCellBuilder, SpCheckBoxPresenter, SpMorphicCheckBoxAdapter, CheckboxMorph and CheckboxButtonMorph, incl. its method drawOnAthensCanvas: from Athens library.
